### ISSUE TITLE
Don't check for `window`

### ISSUE THREAD

**Issue by tjvr** [Hardmath123/nearley#179]
See <https://github.com/Hardmath123/nearley/blob/master/lib/compile.js#L2>:
```js
if (typeof(window) === 'undefined') {
    var nearley = require('../lib/nearley.js');
} else {
    var nearley = window.nearley;
}
```
Some testing frameworks define `window`, so this check fails.

I think there are better conditions we could use here, e.g. the `typeof module !== 'undefined' && typeof module.exports !== 'undefined'` check seen at the bottom of the same file. :)

**Comment by Hardmath123** [comment]
Whoops, must have missed this issue. I don't actually know the Right Thing to do here — I just want the file to work seamlessly in both node and the browser, and ideally to avoid importing some monstrous framework/compiler thing in order to make that happen.

(I wouldn't classify this as "bug", unless it's causing actual issues anywhere.)

**Comment by tjvr** [comment]
> unless it's causing actual issues anywhere

Well it's causing issues for me trying to use `jest` for testing `nearley-reverse`... :P

**Comment by Hardmath123** [comment]
Ah. Maybe we need an `inconvenience` tag instead. :P

But yeah, I'm happy to merge something that fixes this.

**Comment by tjvr** [comment]
We should slap UMD on everything. Yay consistency!

**Comment by tjvr** [comment]
I went ahead and merged #184. Hope that's okay. :)

**Comment by Hardmath123** [comment]
> I went ahead and merged #184. Hope that's okay. :)

Yeah, as long as it doesn't break anything. People do use nearley on the web (for example, http://nearley.js.org/www/quick-compiler.html should hopefully still work as intended).

**Comment by tjvr** [comment]
Yeah! I checked the calculator example html in the repo, and that still seemed to work okay. :)
